stdlib/runtime/malloc.gr

```diff
@@ -174,7 +174,10 @@ export let free = (ap: WasmI32) => {
     // Find the location to insert this block into the free list
     while (true) {
       let nextp = getNext(p)
-      if ( ((blockPtr > p) && (blockPtr < nextp)) || ((p >= nextp) && ((blockPtr > p) || (blockPtr < nextp))) ) {
+      if (
+        blockPtr > p && blockPtr < nextp ||
+        p >= nextp && (blockPtr > p || blockPtr < nextp)
```

Comment:
Is this correct via precedence?

Reply:
Prettier makes this to
`  if (
    (blockPtr > p && blockPtr < nextp) ||
    (p >= nextp && (blockPtr > p || blockPtr < nextp))
  ) {`

Reply:
(extra blank line added my me to make it break like prettier)

Reply:
@marcusroberts I'm wondering about the parens that it removed. Because it removed the ones around `(blockPtr > p && blockPtr < nextp)` and `(p >= nextp && (blockPtr > p || blockPtr < nextp))` but that is very confusing to me—I'm not sure if the logic is correct anymore.

Reply:
Yeah, the logic is correct. `&&` binds tighter than `||`, and we just added the parens for clarity. This one is sort of tough because the AST won't know that you added parens for clarity, and like Marcus said, if we just add parens back then we'll end up with parens everywhere.

I'm actually okay with how this one turned out though—the line break is in a nice place and drives home that it's `blockPtr > p && blockPtr < nextp` or `p >= nextp && (blockPtr > p || blockPtr < nextp)`.